src/diagnostics/MOM_wave_structure.F90

```diff
@@ -0,0 +1,776 @@
+                    Hc(kc-1) = (Hc(kc) + Hc(kc-1))
+                    kc = kc - 1
+                  else ; exit ; endif
+                enddo
+              else
+                ! Add a new layer to the column.
+                kc = kc + 1
+                Rc(kc) = Rf(k,i) ; Hc(kc) = Hf(k,i) 
+              endif
+            enddo
+            ! At this point there are kc layers and the gprimes should be positive.
+            do k=2,kc ! Revisit this if non-Boussinesq.
+              gprime(k) = g_Rho0 * (Rc(k)-Rc(k-1))
+            enddo
+          endif  ! use_EOS?
+          
+          ! Construct and solve tridiagonal system for the interior interfaces
+          ! Note that kc   = number of layers, 
+          !           kc+1 = nzm = number of interfaces, 
+          !           kc-1 = number of interior interfaces (excluding surface and bottom)
+          ! Also, note that "K" refers to an interface, while "k" refers to the layer below.
+          ! Need at least 3 layers (2 internal interfaces) to generate a matrix, also
+          ! need number of layers to be greater than the mode number 
+          if (kc >= 5 .and. kc >= ModeNum + 1) then
+            ! Set depth at surface
+            z_int(1) = 0.0
+            ! Calculate Igu, Igl, depth, and N2 at each interior interface 
+            ! [excludes surface (K=1) and bottom (K=kc+1)] 
+            do K=2,kc
+              Igl(K) = 1.0/(gprime(K)*Hc(k)) ; Igu(K) = 1.0/(gprime(K)*Hc(k-1))
+              z_int(K) = z_int(K-1) + Hc(k-1)
+              N2(K) = gprime(K)/(0.5*(Hc(k)+Hc(k-1)))
+            enddo
+            ! Set stratification for surface and bottom (setting equal to nearest interface for now)
+            N2(1) = N2(2) ; N2(kc+1) = N2(kc) 
+            ! Calcualte depth at bottom
+            z_int(kc+1) = z_int(kc)+Hc(kc)
+            ! check that thicknesses sum to total depth
+            if (abs(z_int(kc+1)-htot(i,j)) > 1.e-10) then
+              call MOM_error(WARNING, "wave_structure: mismatch in total depths")
+              print *, "kc=", kc
+              print *, "z_int(kc+1)=", z_int(kc+1)
+              print *, "htot(i,j)=", htot(i,j)              
+            endif
+
+            ! Populate interior rows of tridiagonal matrix; must multiply through by
+            ! gprime to get tridiagonal matrix to the symmetrical form:
+            ! [-1/H(k-1)]e(k-1) + [1/H(k-1)+1/H(k)-lam_z]e(k) + [-1/H(k)]e(k+1) = 0,
+            ! where lam_z = lam*gprime is now a function of depth.
+            ! Frist, populate interior rows
+            do K=3,kc-1
+              row = K-1 ! indexing for TD matrix rows
+              lam_z(row) = lam*gprime(K)
+              a_diag(row) = gprime(K)*(-Igu(K))
+              b_diag(row) = gprime(K)*(Igu(K)+Igl(K)) - lam_z(row)
+              c_diag(row) = gprime(K)*(-Igl(K))
+              if(isnan(lam_z(row)))then  ; print *, "lam_z(row) is NAN" ; endif
```

Comment:
I don't know how intel was able to compile/link but I have all compilers working now by adding
```
use MOM_checksums, only : isnan => is_NaN
```
which uses our own implementation of the `isnan` function.